src/Serialization/Bech32.purs

```diff
@@ -0,0 +1,36 @@
+module Serialization.Bech32
+  ( Bech32String
+  , verifyBech32
+  , unwrapBech32
+  , getPrefix
+  ) where
+
+import Data.Eq (class Eq)
+import Data.Maybe (Maybe)
+import Data.Show (class Show)
+import FFiHelpers (MaybeFfiHelper, maybeFfiHelper)
+
+foreign import verifyBech32Impl :: MaybeFfiHelper -> Int -> String -> Maybe Bech32String
+foreign import getPrefixImpl :: Int -> Bech32String -> String
+
+-- | Cardano itself doesn't impose limitations on the length of bech32 strings,
+-- but a fixed number must be used in order to use 'bech32.js' and not exceeding
+-- 1023 is strongly suggested.
+bech32Limit :: Int
+bech32Limit = 1023
+
+-- | Represents strings that are in Bech32 BIP173 format but with length limit
+-- | extended to 1023 characters.
+newtype Bech32String = Bech32String String
+
+derive instance Eq Bech32String
+derive newtype instance Show Bech32String
+
+verifyBech32 :: String -> Maybe Bech32String
+verifyBech32 = verifyBech32Impl maybeFfiHelper bech32Limit
```

Comment:
I doubt this approach is what we should follow. This wrapper only guarantees that we have a valid Bech32 data. But what we *really* are interested in is to check if a given string is a valid bech32 encoding of some *particular* data type from serialization-lib.

Here we are pulling a new dependency just to add a new option to fail the decoding which we can't meaningfully handle. Which is completely unnecessary, because `serialization-lib` contains its own bech32 implementation anyway.

Reply:
Even if it does, it doesn't expose it.
In case of Bech32String being just another name for a String i'd give up on the type entirely. WDYT?

Reply:
Yeah a type synonym is enough, as I said in slack.